modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -169,6 +229,174 @@ object ObservationService {
+//      val upImage     = sql"c_image_quality = $image_quality"
+//      val upSky       = sql"c_sky_background = $sky_background"
+//      val upWater     = sql"c_water_vapor = $water_vapor"
+//
+//      val ups: List[AppliedFragment] =
+//        List(
+//          ex.map(upExistence),
+//          subtitle match {
+//            case Nullable.Null  => Some(upSubtitle(None))
+//            case Absent         => None
+//            case NonNull(value) => Some(upSubtitle(Some(value)))
+//          },
+//          status.map(upStatus),
+//          activeState.map(upActive),
+//          constraintSet.toList.flatMap { cs =>
+//            List(
+//              upCloud(cs.cloudExtinction),
+//              upImage(cs.imageQuality),
+//              upSky(cs.skyBackground),
+//              upWater(cs.waterVapor)
+//            )
+//          }
+//        ).flatten
+//
+//      NonEmptyList.fromList(ups).map { nel =>
+//        val up = nel.intercalate(void", ")
+//
+//        import lucuma.core.model.Access._
+//
+//        val where = user.role.access match {
+//
+//          case Service | Admin | Staff =>
+//            sql"""
+//              where o.c_observation_id = $observation_id
+//            """.apply(observationId)
+//
+//          case Ngo => ??? // TODO
+//
+//          case Guest | Pi =>
+//            sql"""
+//              from t_program p
+//              where o.c_observation_id = $observation_id
+//              and o.c_program_id = p.c_program_id
+//              and (
+//                p.c_pi_user_id = $user_id
+//                or
+//                exists(
+//                  select u.c_role
+//                  from   t_program_user u
+//                  where  u.c_program_id = o.c_program_id
+//                  and    u.c_user_id    = $user_id
+//                  and    u.c_role       = 'coi'
+//                )
+//              )
+//            """.apply(observationId ~ user.id ~ user.id)
+//        }
+//
+//        base |+| up |+| where
+//
+//      }
+//    }
+
+    def updateObservations(
+      WHERE:         List[Observation.Id],
+      subtitle:      Nullable[NonEmptyString],
+      ex:            Option[Existence],
+      status:        Option[ObsStatus],
+      activeState:   Option[ObsActiveStatus],
+      constraintSet: Option[ConstraintSet],
+      user:          User
+    ): Option[AppliedFragment] = {
+
+      val base = void"update t_observation o set "
+
+      val upExistence = sql"c_existence = $existence"
+      val upSubtitle  = sql"c_subtitle = ${text_nonempty.opt}"
+      val upStatus    = sql"c_status = $obs_status"
+      val upActive    = sql"c_active_status = $obs_active_status"
+
+      val upCloud     = sql"c_cloud_extinction = $cloud_extinction"
+      val upImage     = sql"c_image_quality = $image_quality"
+      val upSky       = sql"c_sky_background = $sky_background"
+      val upWater     = sql"c_water_vapor = $water_vapor"
```

Comment:
I've also punted on updating the elevation range.  This is an instance where the update you supply has to be consistent with the existing elevation range.  For example, if you set just the minimum airmass then the observation must have a definition already for maximum airmass (as opposed to defining an hour angle range).  I'm not sure how to handle this.  Attempt the update and let the table constraint fail?  Filter out the observations where the update would fail?